lib/ansible/plugins/callback/default.py

```diff
@@ -202,9 +202,9 @@ def _task_start(self, task, prefix=None):
 
         # Preserve task name, as all vars may not be available for templating
         # when we need it later
-        if self._play.strategy == 'free':
-            # Explicitly set to None for strategy 'free' to account for any cached
-            # task title from a previous non-free play
+        if self._play.strategy in ['free', 'host_pinned']:
```

Comment:
The callback doesn't have direct access to the strategy plugin.  Self._play.strategy is just the string loaded from the playbook yaml file.  I'm reading through task_queue_manager.py as it seems like it might make sense to do something in there.  But, if you or anyone else has another idea or I'm not understanding, let me know.

Reply:
the property would have to be attacked to the play as an attribute, but i don't think that is something we should require here, its a long term solution,  the current 'fix' seems good enough until we develop a more general one.

Reply:
to split the baby, make the list of 'free like' strategies a configuration item that users can expand on